core/domain/user_jobs_one_off.py

```diff
@@ -1122,3 +1125,165 @@ def reduce(key, values):
+        exploration_ids = model.exploration_ids
+        exp_rights_models = exp_models.ExplorationRightsModel.get_multi(
+            model.exploration_ids)
+        existing_exploration_ids = [
+            exp_id for exp_id, exp_rights_model
+            in python_utils.ZIP(exploration_ids, exp_rights_models)
+            if exp_rights_model is not None
+        ]
+
+        changed = False
+        if len(existing_exploration_ids) < len(exploration_ids):
+            changed = True
+            yield ('REMOVED_DELETED_EXPS - %s' % class_name, 1)
+
+        if isinstance(model, (
+                user_models.CompletedActivitiesModel,
+                user_models.IncompleteActivitiesModel
+        )):
+            public_exploration_ids = [
+                exp_id for exp_id, exp_rights_model
+                in python_utils.ZIP(exploration_ids, exp_rights_models)
+                if exp_rights_model is not None and
+                exp_rights_model.status == constants.ACTIVITY_STATUS_PUBLIC
+            ]
+
+            if len(public_exploration_ids) < len(existing_exploration_ids):
+                changed = True
+                yield ('REMOVED_PRIVATE_EXPS - %s' % class_name, 1)
+        else:
+            public_exploration_ids = existing_exploration_ids
+
+        if changed:
+            model.exploration_ids = public_exploration_ids
+            model.update_timestamps(update_last_updated_time=False)
+            model.put()
+
+        yield ('SUCCESS - %s' % class_name, 1)
+
+    @staticmethod
+    def reduce(key, values):
+        """Implements the reduce function for this job."""
+        yield (key, len(values))
+
+
+class DeleteNonExistentExpUserDataJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that deletes exploration user data models that do not have
+    existing exploration.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(
+            DeleteNonExistentExpUserDataJob, cls
+        ).enqueue(job_id, shard_count=32)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.ExplorationUserDataModel]
+
+    @staticmethod
+    def map(model):
+        exp_model = exp_models.ExplorationModel.get(
+            model.exploration_id, strict=False)
+        if exp_model is None:
+            exp_services.delete_exploration(
+                feconf.SYSTEM_COMMITTER_ID,
+                model.exploration_id,
+                force_deletion=True
+            )
+            yield ('SUCCESS_DELETED_EXPLORATION', 1)
+        else:
+            yield ('SUCCESS_KEPT', 1)
+
+    @staticmethod
+    def reduce(key, values):
+        """Implements the reduce function for this job."""
+        yield (key, len(values))
+
+
+class DeleteNonExistentExpUserContributionsJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that removes deleted explorations form user contributions models
+    fields.
+    """
```

Comment:
```suggestion
    """Job that removes deleted explorations from UserContributionsModels.
    """
```

Fixed 1 typo & maybe made this a bit clearer.

Reply:
Done.